include/seqan3/utility/char_operations/predicate.hpp

```diff
@@ -55,16 +54,13 @@ inline constexpr auto is_in_interval = detail::is_in_interval_type<interval_firs
  *
  * \details
  *
- * This function like object returns true for all characters of the alphabet, false otherwise.
- * The actual check being performed is whether assigning and then reading a letter results in the original input
- * (but case is ignored).
+ * This function is deprecated and returns always true.
  *
- * ### Example
- * \include test/snippet/utility/char_operations/char_predicate_is_in_alphabet.cpp
  * \hideinitializer
+ * \deprecated Use seqan3::char_is_valid_for instead.
  */
-template <alphabet alphabet_t>
-inline constexpr auto is_in_alphabet = detail::is_in_alphabet_type<alphabet_t>{};
+template <typename alphabet_t>
+SEQAN3_DEPRECATED_310 inline constexpr bool is_in_alphabet = true;
```

Comment:
Oh this is tricky. I see why you did this but this will break the API (and possibly external code!).

There is another solution that is less breaking :)

Revert the changes here except the deprecation warning.
Do not delete detail::is_in_alphabet_type but instead wrap the whole thing in 
```cpp
#ifdef SEQAN3_DEPRECATED_310
 // struct is_in_alphabet_type ...
#endif // SEQAN3_DEPRECATED_310
```

That way, it will remain there without triggering a deprecating warning but we will be able to spot it as "This needs to be deleted once 3.1 is released".

We did that for `seqan3/core/detail/iterator_traits.hpp` if you wanna take a look.

Also do not delete the tests just that but wrap them in 

```cpp

TEST(... non_deprecated)
// ....

#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"

TEST(deprecated_stuff)
// ...

#pragma GCC diagnostic pop
```